src/firewall/iptables.rs

```diff
@@ -189,113 +192,101 @@ impl firewall::FirewallDriver for IptablesDriver {
             setup_portfw.network_name, setup_portfw.container_id
         );
         // Make sure chains exist or create them
-        add_chain_unique(&self.conn, NAT, HOSTPORT_DNAT_CHAIN)?;
-        add_chain_unique(&self.conn, NAT, HOSTPORT_SETMARK_CHAIN)?;
-        add_chain_unique(&self.conn, NAT, NETAVARK_HOSTPORT_MASK_CHAIN)?;
-        add_chain_unique(&self.conn, NAT, &network_dn_chain_name)?;
+        add_chain_unique(conn, NAT, HOSTPORT_DNAT_CHAIN)?;
+        add_chain_unique(conn, NAT, HOSTPORT_SETMARK_CHAIN)?;
+        add_chain_unique(conn, NAT, NETAVARK_HOSTPORT_MASK_CHAIN)?;
+        add_chain_unique(conn, NAT, &network_dn_chain_name)?;
 
         // Setup one-off rules that have nothing to do with ports
         // PREROUTING
         let prerouting_rule = format!("-j {} -m addrtype --dst-type LOCAL", HOSTPORT_DNAT_CHAIN);
-        append_unique(&self.conn, NAT, PREROUTING_CHAIN, &prerouting_rule)?;
+        append_unique(conn, NAT, PREROUTING_CHAIN, &prerouting_rule)?;
 
         // OUTPUT
         let portmap_output_rule =
             format!("-j {} -m addrtype --dst-type LOCAL", HOSTPORT_DNAT_CHAIN);
-        append_unique(&self.conn, NAT, OUTPUT_CHAIN, &portmap_output_rule)?;
+        append_unique(conn, NAT, OUTPUT_CHAIN, &portmap_output_rule)?;
 
         //  SETMARK-CHAIN
         let setmark_rule = format!("-j {}  --set-xmark {}/{}", MARK_JUMP, HEXMARK, HEXMARK);
-        append_unique(&self.conn, NAT, HOSTPORT_SETMARK_CHAIN, &setmark_rule)?;
+        append_unique(conn, NAT, HOSTPORT_SETMARK_CHAIN, &setmark_rule)?;
 
         //  HOSTPORT-MASQ
         let hostport_masq_rule = format!(
             "-j {} -m comment --comment 'netavark portfw masq mark' -m mark --mark {}/{}",
             MASQ_JUMP, HEXMARK, HEXMARK
         );
-        append_unique(
-            &self.conn,
-            NAT,
-            NETAVARK_HOSTPORT_MASK_CHAIN,
-            &hostport_masq_rule,
-        )?;
+        append_unique(conn, NAT, NETAVARK_HOSTPORT_MASK_CHAIN, &hostport_masq_rule)?;
 
         // POSTROUTING
         append_unique(
-            &self.conn,
+            conn,
             NAT,
             POSTROUTING_JUMP,
             &format!("-j {} ", NETAVARK_HOSTPORT_MASK_CHAIN),
         )?;
 
         append_unique(
-            &self.conn,
+            conn,
             NAT,
             POSTROUTING_JUMP,
             &format!(
                 "-j {} -s {} {}",
                 network_chain_name,
-                container_ip.to_string(),
+                setup_portfw.container_ip.to_string(),
                 comment_network_cid
             ),
         )?;
 
         // FOR EACH PORT
-        for i in setup_portfw.port_mappings {
+        for i in setup_portfw.port_mappings.clone() {
+            error!("ipv6 is {}", is_ipv6);
```

Comment:
This should be at least debug

Reply:
of course, this is all just here until we confirm it works.  then i will do more code comments and cleanups